parodusNotifier/parodus_notifier.c

```diff
@@ -0,0 +1,149 @@
+    }
+
+    NotifPrint("parodus url formed is %s\n", *url);
+}
+
+int send_event_to_parodus(notifier_t *msg)
+{
+    char *parodus_url = NULL;
+    libpd_instance_t client_instance;
+    int ret = -1;
+    wrp_msg_t *notif_wrp_msg = NULL;
+
+    get_parodus_url(&parodus_url);
+    NotifPrint("parodus_url is %s\n", parodus_url);
+    if(parodus_url != NULL && msg->service_name != NULL )
+    {
+        NotifPrint("service_name is %s\n", msg->service_name);
+        libpd_cfg_t cfg = {.service_name = msg->service_name,
+						.receive = false, .keepalive_timeout_secs = 0,
+						.parodus_url = parodus_url,
+						.client_url = NULL
+					   };
+
+         ret = libparodus_init (&client_instance, &cfg);
+         if(ret == 0)
+         {
+            NotifInfo("Init for parodus Success..!!\n");
+            if(msg->source != NULL && msg->destination != NULL && msg->payload != NULL && msg->content_type != NULL)
+            {
+                NotifPrint("source: %s destination: %s payload: %s content_type: %s\n",msg->source, msg->destination, msg->payload, msg->content_type);
+                notif_wrp_msg = (wrp_msg_t *)malloc(sizeof(wrp_msg_t));
+                if(notif_wrp_msg != NULL)
+                {
+                    memset(notif_wrp_msg, 0, sizeof(wrp_msg_t));
+                    notif_wrp_msg ->msg_type = WRP_MSG_TYPE__EVENT;
+                    notif_wrp_msg ->u.event.source = msg->source;
+                    notif_wrp_msg ->u.event.dest = msg->destination;
+                    notif_wrp_msg->u.event.content_type = msg->content_type;
+                    notif_wrp_msg ->u.event.payload = (void *)(msg->payload);
+                    notif_wrp_msg ->u.event.payload_size = strlen(notif_wrp_msg ->u.event.payload);
+                    ret = libparodus_send(client_instance, notif_wrp_msg );
+                    if(ret == 0)
+                    {
+                        NotifInfo("Notification successfully sent to parodus\n");
+                    }
+                    wrp_free_struct (notif_wrp_msg);
+                }
+            }
+         }
+    }
+    return ret;
+}
+
+void free_notifier_struct(notifier_t *msg)
+{
+    if(msg != NULL)
+    {
+        if(msg->source != NULL)
+        {
+            free(msg->source);
+            msg->source = NULL;
+        }
+
+        if(msg->destination != NULL)
+        {
+            free(msg->destination);
+            msg->destination = NULL;
+        }
+
+        if(msg->service_name != NULL)
+        {
+            free(msg->service_name);
+            msg->service_name = NULL;
+        }
+
+        if(msg->content_type != NULL)
+        {
+            free(msg->content_type);
+            msg->content_type = NULL;
+        }
+
+        if(msg->payload != NULL)
+        {
+            free(msg->payload);
+            msg->payload = NULL;
+        }
+        free(msg);
+        msg = NULL;
+    }
+}
+
+const char *rdk_logger_module_fetch(void)
+{
+    return "LOG.RDK.PARODUSNOTIFIER";
```

Comment:
Library cannot have its own logger, so accept this as argument.
For cli we should not need any logger.